### PR TITLE
Bugfix/username change

### PR DESCRIPTION
**Pull request by seripap**
Actually checking username length and forcing letter first standard.

### CODE REVIEW

**Comment by seripap** [comment]
Ignore the deployed to heroku- it's not running this branch until the container fix branch gets merged in.

File: src/js/chat.js [hunk 1]
```diff
@@ -68,9 +72,10 @@ export default class Chat {
     const COLORS = [
       '#e21400', '#ffe400', '#ff8f00',
       '#58dc00', '#dd9cff', '#4ae8c4',
-      '#3b88eb', '#f47777', '#d300e7',
-      '#99FF33', '#99CC33', '#999933',
-      '#996633', '#993333', '#990033',
+      '#3b88eb', '#f47777', '#EEACB7',
```
Comment (alanfriedman) [inline src/js/chat.js]: We should check that all of these colors are legible on black background.

File: src/js/darkwire.js [hunk 2]
```diff
@@ -105,46 +101,78 @@ export default class Darkwire {
     return importKeysPromises;
   }
 
+  createUser(resolve, reject, username) {
+    Promise.all([
```
Comment (alanfriedman) [inline src/js/darkwire.js]: I think this is leftover but Promise.all isn't needed for just one promise so we can remove it.

File: src/js/darkwire.js [hunk 3]
```diff
@@ -105,46 +101,78 @@ export default class Darkwire {
...
+      this._keys = {
+        public: data[0].publicKey,
+        private: data[0].privateKey
+      };
+      return Promise.all([
+        this._cryptoUtil.exportKey(data[0].publicKey, 'spki')
+      ]);
+    })
+    .then((exportedKeys) => {
+      if (!exportedKeys) {
```
Comment (alanfriedman) [inline src/js/darkwire.js]: Why would this be null?

Comment (seripap) [inline src/js/darkwire.js]: Something could have gone wrong?

File: src/js/darkwire.js [hunk 4]
```diff
@@ -105,46 +101,78 @@ export default class Darkwire {
...
         }
+
+        return resolve({
+          username: username,
+          publicKey: user.publicKey
+        });
       }
+
+      // User doesn't exist, create the user
+      return this.createUser(resolve, reject, username);
```
Comment (alanfriedman) [inline src/js/darkwire.js]: I don't get this. Why do you create a new user?

File: src/js/darkwire.js [hunk 5]
```diff
@@ -105,46 +101,78 @@ export default class Darkwire {
...
+    return false;
+  }
+
   removeUser(data) {
     this._users = _.without(this._users, _.findWhere(this._users, {id: data.id}));
     return this._users;
   }
 
-  updateUsername(username, newUsername) {
+  updateUsername(username) {
```
Comment (alanfriedman) [inline src/js/darkwire.js]: Are we sanitizing usernames? Is it possible to change your name to "<script>alert('hax0r');</script>" ?

File: src/js/darkwire.js (hunk 4 above)
Comment (seripap) [inline src/js/darkwire.js]: This happens on user join. It should actually be a promise

File: src/js/darkwire.js (hunk 5 above)
Comment (seripap) [inline src/js/darkwire.js]: Happens [before].(https://github.com/seripap/darkwire.io/blob/bugfix/usernameChange/src/js/chat.js#L168-L180)

Comment (alanfriedman) [inline src/js/darkwire.js]: Ok - seems kinda weird to use `updateUsername` at the start but we can tweak it later.

**Pull request merged**